Training/Training/file.c

```diff
@@ -1,13 +1,154 @@
-// ------------------------------------------------------------------------------------------------
+// --------------------------------------------------------------------------------------------------------
 // Training ~ A training program for new joiners at Metamation, Batch - July 2024.
-// Copyright (c) Metamation India.
+// Copyright (c) Gokul TS @Metamation India.
 // ------------------------------------------------------------------
-// Program.c
-// Program on main branch.
-// ------------------------------------------------------------------------------------------------
-
+// file.c
+// C Program to check test cases
+// --------------------------------------------------------------------------------------------------------
+#pragma warning(disable:6385)
+#include "SortAndSearch.h"
 #include <stdio.h>
+#include <string.h>
+#include <ctype.h>
+#include <stdlib.h>
+#include <malloc.h>
+#include <time.h>
+
+/// <summary>Gets integer input from user</summary>
+/// <param name="prompt"></param>
+int GetUserInput (char* prompt);
+/// <summary>Implements the search and sort algorithm</summary>
+void SortAndSearch (void(*sortFunc)(int[], int), int (*searchFunc)(int[], int, int));
+/// <summary>Checks test cases for Sort function</summary>
+static void TestCase_Sort (void(*func)(int[], int));
+/// <summary>Checks test cases for Search function</summary>
+static void TestCase_Search (int (*func)(int[], int, int));
+/// <summary>Returns whether the given array is sorted or not</summary>
+int IsSorted (int* arr1, int size);
+/// <summary>Clears the content on screen</summary>
+void ClrScreen ();
+
+int main () {
+   while (1) {
+      int num = GetUserInput ("1-INSERTION SORT\n2-TEST CASES (InsertionSort)\n3-TEST CASES (BinarySearch)"
+                              "\n4-CLR SCREEN\n5-EXIT\nEnter an option:");
+      switch (num) {
+         case 1:
+            SortAndSearch (InsertionSort,BinarySearch);
+            break;
+         case 2:
+            TestCase_Sort (InsertionSort);
+            break;
+         case 3:
+            TestCase_Search (BinarySearch);
+            break;
+         case 4:
+            ClrScreen ();
+            break;
+         case 5:
+            return 0;
+         default:
+            printf (ANSI_COLOR_RED"Please enter a number between 1 - 6\n"ANSI_RESET_ALL);
+      }
+   }
+   return 0;
+}
+
+int GetUserInput (char* prompt) {
+   int number = 0;
+   if (prompt != "") printf ("%s", prompt);
+   for (;;) {
+      char strNum[MAX_DIGIT], * endPtr = NULL,
+         * strPtr = fgets (strNum, sizeof (strNum), stdin),
+         * validInput = strchr (strNum, '\n');
+      number = strtol (strNum, &endPtr, 10);
+      if (validInput == NULL)  while (getchar () != '\n');
+      if (strPtr == NULL || (!isdigit (strNum[0]) && strNum[0] != '-') || *endPtr != '\n') {
+         printf (ANSI_COLOR_RED"Invalid Number!!! Please re-enter\n"ANSI_RESET_ALL);
+         continue;
+      }
+      break;
```

Comment:
Since you would like for the control to pass through the loop at least once, a do-while loop seems to be the more elegant choice.